src/java/fr/paris/lutece/plugins/announce/service/announcesearch/AnnounceSearchService.java

```diff
@@ -263,15 +263,9 @@ public String processIndexing( boolean bCreate )
 
             boolean bIsLocked = false;
 
-            if ( IndexWriter.isLocked( dir ) )
```

Comment:
add a test for knowing if the index is already use

Reply:
With the addition of a critical code block (synchronized), which utilizes a lock to control concurrent access, access to the IndexWriter directory is now managed securely, eliminating the need to test for concurrent access.